externs/shaka/manifest.js

```diff
@@ -342,6 +347,11 @@ shaka.extern.CreateSegmentIndexFunction;
  *   as the value. If the channel number is not provided by the description,
  *   we'll set an 0-based index as the key.
  *   Example: {'CC1': 'eng'; 'CC3': 'swe'}, or {'1', 'eng'; '2': 'swe'}, etc.
+ * @property {(string|undefined)} tilesLayout
```

Comment:
I have been thinking about how to do this and the truth is that I am not very clear about it (due to the large number of changes that it takes to support: https://github.com/google/shaka-player/issues/2429).

In DASH the layout is "provided" at the playlist level, in HLS it is done at the segment level.

Changing this in the future is not a problem because the API that is exposed does not need to know the layout, all the logic is done in the `getThumbnails` function.

Note: for HLS I do not intend to send a PR as none of our providers support the image format in HLS due to it not being an official spec (https://tools.ietf.org/html/draft-pantos- http-live-streaming-23)

What do you think?

Reply:
I might be asking the same thing as Joey, but why is this a string if it's essentially a matrix? Because this is the way it's represented in DASH?

Reply:
I want to leave a string because it is the original value that comes in DASH (this is also valid for HLS). This value is what was originally represented.